src/sickle.ts

```diff
@@ -399,6 +406,75 @@ class Annotator {
     }
   }
 
+  private visitExterns(node: ts.Node, namespace: string[] = []) {
+    this.currentOutput = this.externsOutput;
+    switch (node.kind) {
+      case ts.SyntaxKind.ModuleDeclaration:
+        let decl = <ts.ModuleDeclaration>node;
+        namespace = namespace.concat([decl.name.text]);
```

Comment:
I don't want to mutate the array, because I need the element to pop back off when I'm done (in the cases where I recurse multiple times).